src/promptflow/promptflow/_sdk/_serving/app.py

```diff
@@ -53,6 +50,14 @@ def init(self, **kwargs):
             logger.info(f"Project path: {self.project_path}")
             self.flow_entity = load_flow(self.project_path)
             self.flow = self.flow_entity._init_executable()
+
+            # enable environment_variables
+            self.environment_variables = kwargs.get("environment_variables", {})
+            os.environ.update(self.environment_variables)
+            default_environment_variables = self.flow.get_environment_variables_with_overrides()
+            self.set_default_environment_variables(default_environment_variables)
+            logger.info(f"Environment variable keys: {self.environment_variables.keys()}")
+
```

Comment:
logger level won't take effect in serving because the logger is initialized before this env update,  maybe we should try to update the logger level after this env update if it's set in the flow environment variables.